pkg/scalers/stan_scaler.go

```diff
@@ -278,6 +278,7 @@ func (s *stanScaler) GetMetricsAndActivity(ctx context.Context, metricName strin
 	}
 	totalLag := s.getMaxMsgLag()
 	s.logger.V(1).Info("Stan scaler: Providing metrics based on totalLag, threshold", "totalLag", totalLag, "lagThreshold", s.metadata.lagThreshold)
+	s.logger.Info("The Stan scaler (NATS Streaming) is DEPRECATED and will be removed in v2.18 - Use scaler 'nats-jetstream' instead")
```

Comment:
nit: given 2.16 has already been released and this deprecation notice will be published in 2.17, I think technically this should be 2.19 (not that it matters much because upstream has deprecated this in [2023](https://nats-io.gitbook.io/legacy-nats-docs/nats-streaming-server-aka-stan))

Reply:
Yeah, you are right. Let's keep the line correct :) I'll adjust this.

Reply:
fixed :)